src/main/java/org/jabref/gui/libraryproperties/LibraryPropertiesDialogView.java

```diff
@@ -0,0 +1,214 @@
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(btn -> {
+            if (btn == ButtonType.OK) {
+                storeSettings();
+            }
+            return null;
+        });
+
+        setTitle(Localization.lang("Library properties"));
+
+    }
+
+    @FXML
+    private void initialize() {
+        viewModel = new LibraryPropertiesDialogViewModel(panel, dialogService, preferencesService);
+
+        generalFileDirectory.textProperty().bindBidirectional(viewModel.generalFileDirectoryPropertyProperty());
+        userSpecificFileDirectory.textProperty().bindBidirectional(viewModel.userSpecificFileDirectoryProperty());
+
+        encoding.itemsProperty().bind(viewModel.encodingsProperty());
+        encoding.valueProperty().bindBidirectional(viewModel.selectedEncodingProperty());
+        saveOrderPanel = new SaveOrderConfigDisplay();
+        saveOrdervbox.getChildren().add(saveOrderPanel.getJFXPanel());
+        fieldFormatterCleanupsPanel = new FieldFormatterCleanupsPanel(Localization.lang("Enable save actions"),
+                                                                      Cleanups.DEFAULT_SAVE_ACTIONS);
+
+        contentVbox.getChildren().add(fieldFormatterCleanupsPanel);
+        saveInOriginalOrder.selectedProperty().bindBidirectional(viewModel.saveInOriginalProperty());
+        saveInSpecifiedOrder.selectedProperty().bindBidirectional(viewModel.saveInSpecifiedOrderProperty());
+
+        protect.selectedProperty().bindBidirectional(viewModel.libraryProtectedProperty());
+        saveOrderPanel.getJFXPanel().disableProperty().bind(viewModel.saveInOriginalProperty());
+
+        setValues();
+        updateEnableStatus();
+    }
+
+    private void setValues() {
+
+        viewModel.setValues();
+        Optional<SaveOrderConfig> storedSaveOrderConfig = panel.getBibDatabaseContext().getMetaData().getSaveOrderConfig();
+        boolean selected;
+        if (!storedSaveOrderConfig.isPresent()) {
+            viewModel.saveInOriginalProperty().setValue(true);
+            oldSaveOrderConfig = SaveOrderConfig.getDefaultSaveOrder();
+            selected = false;
+        } else {
+            SaveOrderConfig saveOrderConfig = storedSaveOrderConfig.get();
+            oldSaveOrderConfig = saveOrderConfig;
+            if (saveOrderConfig.saveInOriginalOrder()) {
+                viewModel.saveInOriginalProperty().setValue(true);
+                selected = false;
+            } else {
+                viewModel.saveInSpecifiedOrderProperty().setValue(true);
+                selected = true;
+            }
+            saveOrderPanel.setSaveOrderConfig(saveOrderConfig);
+
+        }
+        saveOrderPanel.setEnabled(selected);
+
+        fieldFormatterCleanupsPanel.setValues(panel.getBibDatabaseContext().getMetaData());
+    }
+
+    public void updateEnableStatus() {
+        DatabaseLocation location = panel.getBibDatabaseContext().getLocation();
```

Comment:
this should be controlled by the viewmodel as well (and be achieved via bindings).